# Perfil de App Gratuitos na Play Story e na App Story

Esse projeto pretende-se encontrar o perfis de aplativos gratuitos nas lojas das Play Story e da App Story que sejam  rentáveis para uma empresa. Essa empresa desenvolve aplicativo gratuitos e sua principal fonte de receitas é as visualizações de anuncios que são mostrados no aplicativos.

Portanto, nosso objetivo desse projeto, é analisar os dados disponiveis e extrair informações a respeito dos habitos dos usários para ajudar os desenvolvedores a traçar estrategias que atraiam o maior número de usários possível.

## Abrindo nossa base da dados

Usaremos duas aquivos para fazer essa analis. Um arquivo contendo os dados da loja da Play store e outro arquivo da loja da Apple Stores.  Os dois arquivos estão disponíveis nos links a baixo:
 * [DataSet Play Story](https://www.kaggle.com/datasets/lava18/google-play-store-apps)
 * [DataSet Apple Story](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps)

In [54]:
from csv import reader # importação das bibliotecas  utilizadas 

In [58]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
playStory = list(read_file)
playStory_header = playStory[0]
playStory = playStory[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
appleStory = list(read_file)
appleStory_header = appleStory[0]
appleStory = appleStory[1:]

Para facilitar a nossa visualização de dados vamos criar uma função explore_data() que imprime as linhas do dataset de maneira legível. Além disso, essa função também mostra o número de colunas existente no dataSet e o seu número de linha.

In [59]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Numero de linhas:', len(dataset))
        print('Number de colunas:', len(dataset[0]))

print(playStory_header)
print('\n')
explore_data(playStory, 0,2 , True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


Podemos observar que o  dataset da loja do PlayStory tem  13 colunas com  10.841 linhas

In [10]:
print(appleStory_header)
print('\n')
explore_data(appleStory, 0,2 , True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 7197
Number of columns: 16


Já  o dataset da Apple Story tem 16 colunas e 7197 linhas

## Excluindo dados errados

O conjunto de dados da Play Story tem uma seção dedicada que descreve um erro na linha  10472. Vamos imprimir o cabeçalho do dataset, a linha com erro e a uma linha sem erro  para entendemos melhor o problema!

In [12]:
print(playStory[10472])  
print('\n')
print(playStory_header)  
print('\n')
print(playStory[0])      

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


A linha 10472 corresponde  ao aplicativo Life Made WI-Fi Touchscreen Photo Frame. Se contem dois erros. O primeiro é classificação ("rating") que está com o valor 19, entranto o valor máximo de classificação no Play Story é 5.0. Outro problema é a falta do valor está no valor categoria ("category"). Como solução vamos excluir essa linha do dataSet.

In [13]:
print(len(playStory))
del playStory[10472]  # don't run this more than once
print(len(playStory))

10841
10840


## Removendo entradas duplicadas

Quando analisamos o dataSet a fundo, percebemos que  existem aplicativos com mais de uma entrada, como por exemplo o Intagram que possui 4 entradas.

In [14]:
for app in playStory:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


A um total de 1181 casos em que o aplicativo tem uma entrada duplicada 

In [17]:
duplicados_apps = []
unico_apps = []

for app in playStory:
    name = app[0]
    if name in unico_apps:
        duplicados_apps.append(name)
    else:
        unico_apps.append(name)
    
print('O Número de Aplicativos duplicados é:', len(duplicados_apps))
print('\n')
print('Examplo de aplicativo duplicado:', duplicados_apps[:15])

O Número de Aplicativos duplicados é: 1181


Examplo de aplicativo duplicado: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Não queremos analisar aplicativos mais de uma vez, se fizemos isso nossa analisar estará inadequada. Portanto, precimos deletar os entradas que são duplicadas a e deixar apenas uma unica entrada por aplicativo.

Quando observações entradas do Instagram, percebomos que elas têm diferentes valores de avaliação ("Reviews"). Então no lugar de excluir  essas entradas aletóriamente vamos manter a entrada que tem o maior número de avaliaçãoe e deletar o restante,  considerando que quanto maior o número de avaliação maior confiável  será classificações.

In [19]:
avaliacao_max = {}

for app in playStory:
    name = app[0]
    n_avaliacao = float(app[3])
    
    if name in avaliacao_max and avaliacao_max[name] < n_avaliacao:
        avaliacao_max[name] = n_avaliacao
        
    elif name not in avaliacao_max:
        avaliacao_max[name] = n_avaliacao

Sabemos que existem 1181 casos de aplicativos com entradas duplicadas, então o dicionário avaliacao_max, precisa ter o tamanho do dataSet - 1181. Vejamos se isso é verdade:

In [21]:
print('Expecativa do tamanho:', len(playStory) - 1181)
print('Tamanho Atual:', len(avaliacao_max))

Expecativa do tamanho: 9659
Tamanho Atual: 9659


Agora, vamos usar o dicionário avaliacao para deletar os dados duplicados. Para os casos duplicados, manteremos apenas as entradas com o maior número de avaliações.

In [24]:
playStory_clean = []
adicionado = []

for app in playStory:
    name = app[0]
    n_avaliacao = float(app[3])
    
    if (avaliacao_max[name] == n_avaliacao) and (name not in adicionado):
        playStory_clean.append(app)
        adicionado.append(name) 

Agora vamos ver se nossa limpeza deu certo. 

In [26]:
explore_data(playStory_clean, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9659
Number of columns: 13


Agora nosso dataSet tem 9659 linhas como já era esperado

## Removendo aplicativos que não são em inglês

No DataSet existe um conjunto de dados que não são voltados para o publicos que fala inglês. Veja ai abaixo esse conjunto de dados:


In [28]:
print(appleStory[813][1])
print(appleStory[6731][1])

print(playStory_clean[4412][0])
print(playStory_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


Não estamos interessados em manter esse tipo de aplicativo, então vamos removê-los. Uma maneira de fazer isso é remover cada aplicativo cujo nome contém um símbolo que não é comumente usado em texto em inglês – o texto em inglês geralmente inclui letras do alfabeto inglês, números compostos por dígitos de 0 a 9, sinais de pontuação (., ! , ?, ;, etc.) e outros símbolos (+, *, /, etc.).

Todos esses caracteres específicos para textos em inglês são codificados usando o padrão ASCII. Cada caractere ASCII tem um número correspondente entre 0 e 127 associado a ele, e podemos aproveitar isso para construir uma função que verifica o nome de um aplicativo e nos informa se ele contém caracteres não ASCII.

Construímos esta função abaixo e usamos a função `ord()` integrada para descobrir o número de codificação correspondente de cada caractere

In [29]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


A função parece funcionar bem, mas alguns nomes de aplicativos em inglês usam emojis ou outros símbolos (™, — (em dash), – (en dash), etc.) que estão fora do intervalo ASCII. Por isso, removeremos aplicativos úteis se usarmos a função em sua forma atual.

In [30]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


Para minimizar o impacto da perda de dados, só removeremos um aplicativo se seu nome tiver mais de três caracteres não ASCII:

In [31]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


A função ainda não é perfeita, e pouquíssimos aplicativos em inglês podem passar pelo nosso filtro, mas isso parece bom o suficiente neste ponto de nossa análise - não devemos gastar muito tempo em otimização neste momento.

Abaixo, usamos a função `is_english()` para filtrar os aplicativos que não estão em inglês para ambos os conjuntos de dados:

In [32]:
playStory_english = []
appleStory_english = []

for app in playStory_clean:
    name = app[0]
    if is_english(name):
        playStory_english.append(app)
        
for app in appleStory:
    name = app[1]
    if is_english(name):
        appleStory_english.append(app)
        
explore_data(playStory_english, 0, 3, True)
print('\n')
explore_data(appleStory_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

Podemos ver que ficamos com 9614 aplicativos Android e 6183 aplicativos iOS.

## Isolando os aplicativos gratuitos

Como mencionamos na introdução, criamos apenas aplicativos gratuitos para download e instalação, e nossa principal fonte de receita consiste em anúncios no aplicativo. Nossos conjuntos de dados contêm aplicativos gratuitos e não gratuitos, e precisaremos isolar apenas os aplicativos gratuitos para nossa análise. Abaixo, isolamos os aplicativos gratuitos para ambos os conjuntos de dados.

In [34]:
playStory_final = []
appleStory_final = []

for app in playStory_english:
    price = app[7]
    if price == '0':
        playStory_final.append(app)
        
for app in appleStory_english:
    price = app[4]
    if price == '0.0':
        appleStory_final.append(app)
        
print(len(playStory_final))
print(len(appleStory_final))

8864
3222


Ficamos com 8.864 aplicativos Android e 3.222 aplicativos iOS, o que deve ser suficiente para nossa análise.

## Aplicativos mais comuns por gênero

### Parte I

Como mencionamos na introdução, nosso objetivo é determinar os tipos de aplicativos que provavelmente atrairão mais usuários, pois nossa receita é altamente influenciada pelo número de pessoas que usam nossos aplicativos.

Para minimizar riscos e sobrecarga, nossa estratégia de validação para uma ideia de aplicativo é composta por três etapas:
262 / 5.000
Resultados de tradução
1. Crie uma versão mínima do aplicativo para Android e adicione-a ao Google Play.
2. Se o aplicativo tiver uma boa resposta dos usuários, nós o desenvolvemos ainda mais.
3. Se o aplicativo for lucrativo após seis meses, também criamos uma versão iOS do aplicativo e a adicionamos à App Store.

Como nosso objetivo final é adicionar o aplicativo na App Store e no Google Play, precisamos encontrar perfis de aplicativos que sejam bem-sucedidos em ambos os mercados. Por exemplo, um perfil que pode funcionar bem para ambos os mercados pode ser um aplicativo de produtividade que faz uso de gamificação.

Vamos começar a análise entendendo os gêneros mais comuns para cada mercado. Para isso, criaremos uma tabela de frequência para a coluna `prime_genre` do conjunto de dados da App Store e as colunas `Genres` e `Category` do conjunto de dados do Google Play.

Parte II

Construiremos duas funções que podemos usar para analisar as tabelas de frequência:

- Uma função para gerar tabelas de frequência que mostram porcentagens
- Outra função que podemos usar para exibir as porcentagens em ordem decrescente

In [35]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [36]:
display_table(appleStory_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Podemos ver que entre os aplicativos gratuitos em inglês, mais da metade (58,16%) são jogos. Os aplicativos de entretenimento estão perto de 8%, seguidos pelos aplicativos de foto e vídeo, que estão perto de 5%. Apenas 3,66% dos aplicativos são projetados para educação, seguidos por aplicativos de redes sociais que representam 3,29% dos aplicativos em nosso conjunto de dados.

A impressão geral é que a App Store (pelo menos a parte que contém aplicativos gratuitos em inglês) é dominada por aplicativos que são projetados para diversão (jogos, entretenimento, foto e vídeo, redes sociais, esportes, música etc.), enquanto aplicativos com práticas propósitos (educação, compras, serviços públicos, produtividade, estilo de vida, etc.) são mais raros. No entanto, o fato de os aplicativos divertidos serem os mais numerosos também não implica que eles também tenham o maior número de usuários – a demanda pode não ser a mesma da oferta.

Vamos continuar examinando as colunas `Genres` e `Category` do conjunto de dados do Google Play (duas colunas que parecem estar relacionadas).

In [38]:
display_table(playStory_final, 1) # Categoria

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

O cenário parece significativamente diferente no Google Play: não há muitos aplicativos projetados para diversão e parece que um bom número de aplicativos é projetado para fins práticos (família, ferramentas, negócios, estilo de vida, produtividade etc.). No entanto, se investigarmos mais a fundo, podemos ver que a categoria família (que responde por quase 19% dos aplicativos) significa principalmente jogos para crianças.





[![img](https://s3.amazonaws.com/dq-content/350/py1m8_family.png)](https://play.google.com/store/apps/category/FAMILY?hl=en)


Mesmo assim, aplicativos práticos parecem ter uma representação melhor no Google Play em comparação com a App Store. Esta imagem também é confirmada pela tabela de frequências que vemos para a coluna `Genres`:

In [39]:
display_table(playStory_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

A diferença entre as colunas `Gêneros` e `Categoria` não é muito clara, mas uma coisa que podemos notar é que a coluna `Gêneros` é muito mais granular (tem mais categorias). No momento, estamos apenas procurando uma visão geral, então trabalharemos apenas com a coluna 'Categoria' daqui para frente.

Até este ponto, descobrimos que a App Store é dominada por aplicativos projetados para diversão, enquanto o Google Play mostra um cenário mais equilibrado de aplicativos práticos e divertidos. Agora gostaríamos de ter uma ideia sobre o tipo de aplicativo que tem mais usuários.

## Aplicativos mais populares por gênero na App Store

Uma maneira de descobrir quais gêneros são os mais populares (têm mais usuários) é calcular o número médio de instalações para cada gênero de aplicativo. Para o conjunto de dados do Google Play, podemos encontrar essas informações na coluna Instalações, mas para o conjunto de dados da App Store, essas informações estão ausentes. Como solução alternativa, usaremos o número total de avaliações de usuários como proxy, que podemos encontrar no aplicativo rating_count_tot.

Abaixo, calculamos o número médio de avaliações de usuários por gênero de aplicativo na App Store:

In [41]:
genres_appleStory = freq_table(appleStory_final, -5)

for genre in genres_appleStory:
    total = 0
    len_genre = 0
    for app in appleStory_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Em média, os aplicativos de navegação têm o maior número de comentários de usuários, mas esse número é fortemente influenciado pelo Waze e pelo Google Maps, que têm cerca de meio milhão de comentários de usuários juntos:

In [42]:
for app in appleStory_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


O mesmo padrão se aplica a aplicativos de redes sociais, onde o número médio é fortemente influenciado por alguns gigantes como Facebook, Pinterest, Skype, etc. O mesmo se aplica a aplicativos de música, onde alguns grandes players como Pandora, Spotify e Shazam influenciam fortemente o número médio.

Nosso objetivo é encontrar gêneros populares, mas aplicativos de navegação, redes sociais ou música podem parecer mais populares do que realmente são. O número médio de classificações parece ser distorcido por muito poucos aplicativos que têm centenas de milhares de classificações de usuários, enquanto os outros aplicativos podem ter dificuldades para ultrapassar o limite de 10.000. Poderíamos obter uma imagem melhor removendo esses aplicativos extremamente populares para cada gênero e depois retrabalhando as médias, mas deixaremos esse nível de detalhe para mais tarde.

Os aplicativos de referência têm 74.942 classificações de usuários em média, mas na verdade são a Bíblia e o Dictionary.com que distorcem a classificação média:

In [43]:
for app in appleStory_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


No entanto, este nicho parece mostrar algum potencial. Uma coisa que poderíamos fazer é pegar outro livro popular e transformá-lo em um aplicativo onde poderíamos adicionar recursos diferentes além da versão bruta do livro. Isso pode incluir citações diárias do livro, uma versão em áudio do livro, questionários sobre o livro etc. Além disso, também podemos incorporar um dicionário no aplicativo, para que os usuários não precisem sair do aplicativo para procurar até palavras em um aplicativo externo.

Essa ideia parece se encaixar bem com o fato de que a App Store é dominada por aplicativos para diversão. Isso sugere que o mercado pode estar um pouco saturado com aplicativos para diversão, o que significa que um aplicativo prático pode ter mais chance de se destacar entre o grande número de aplicativos na App Store.

Outros gêneros que parecem populares incluem clima, livro, comida e bebida ou finanças. O gênero livro parece se sobrepor um pouco à ideia do aplicativo que descrevemos acima, mas os outros gêneros não nos parecem muito interessantes:

- Aplicativos de clima — as pessoas geralmente não passam muito tempo no aplicativo e as chances de lucrar com os anúncios no aplicativo são baixas. Além disso, obter dados meteorológicos confiáveis ​​​​ao vivo pode exigir que conectemos nossos aplicativos a APIs não gratuitas.

- Comida e bebida — os exemplos aqui incluem Starbucks, Dunkin' Donuts, McDonald's, etc. Portanto, criar um aplicativo popular de comida e bebida requer culinária real e um serviço de entrega, que está fora do escopo de nossa empresa.

- Aplicativos financeiros — esses aplicativos envolvem serviços bancários, pagamento de contas, transferência de dinheiro etc. Criar um aplicativo financeiro requer conhecimento de domínio e não queremos contratar um especialista em finanças apenas para criar um aplicativo.

Agora vamos analisar um pouco o mercado do Google Play.

## Aplicativos mais populares por gênero no Google Play

Para o mercado do Google Play, na verdade temos dados sobre o número de instalações, então podemos ter uma visão mais clara sobre a popularidade do gênero. No entanto, os números de instalação não parecem precisos o suficiente - podemos ver que a maioria dos valores é aberta (100+, 1.000+, 5.000+, etc.):

In [45]:
display_table(playStory_final, 5) # As colunas de instalação

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


Um problema com esses dados é que não são precisos. Por exemplo, não sabemos se um aplicativo com mais de 100.000 instalações tem 100.000 instalações, 200.000 ou 350.000. No entanto, não precisamos de dados muito precisos para nossos propósitos - queremos apenas ter uma ideia de quais gêneros de aplicativos atraem mais usuários e não precisamos de precisão perfeita em relação ao número de usuários.

Vamos deixar os números como estão, o que significa que vamos considerar que um aplicativo com mais de 100.000 instalações possui 100.000 instalações e um aplicativo com mais de 1.000.000 instalações possui 1.000.000 instalações e assim por diante.

Para realizar cálculos, no entanto, precisaremos converter cada número de instalação para `float` — isso significa que precisamos remover as vírgulas e os caracteres de adição, caso contrário a conversão falhará e gerará um erro. Faremos isso diretamente no loop abaixo, onde também calculamos o número médio de instalações para cada gênero (categoria).

In [48]:
categories_playStory = freq_table(playStory_final, 1)

for category in categories_playStory:
    total = 0
    len_category = 0
    for app in playStory_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Em média, os aplicativos de comunicação têm mais instalações: 38.456.119. Esse número é fortemente distorcido por alguns aplicativos com mais de um bilhão de instalações (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail e Hangouts) e alguns outros com mais de 100 e 500 milhões de instalações:

In [49]:
for app in playStory_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Se removêssemos todos os aplicativos de comunicação com mais de 100 milhões de instalações, a média seria reduzida cerca de dez vezes:

In [60]:
under_100_m = []

for app in playStory_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

Vemos o mesmo padrão para a categoria de players de vídeo, que é a segunda colocada com 24.727.872 instalações. O mercado é dominado por aplicativos como Youtube, Google Play Movies & TV ou MX Player. O padrão se repete para aplicativos sociais (onde temos gigantes como Facebook, Instagram, Google+, etc.), aplicativos de fotografia (Google Fotos e outros editores de fotos populares) ou aplicativos de produtividade (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.) .).

Novamente, a principal preocupação é que esses gêneros de aplicativos possam parecer mais populares do que realmente são. Além disso, esses nichos parecem ser dominados por alguns gigantes com os quais é difícil competir.

O gênero de jogo parece bastante popular, mas anteriormente descobrimos que essa parte do mercado parece um pouco saturada, então gostaríamos de sugerir uma recomendação de aplicativo diferente, se possível.

Os livros e o gênero de referência também parecem bastante populares, com um número médio de instalações de 8.767.811. É interessante explorar isso com mais profundidade, pois descobrimos que esse gênero tem potencial para funcionar bem na App Store, e nosso objetivo é recomendar um gênero de aplicativo que mostre potencial para ser lucrativo tanto na App Store quanto no Google Play.

Vamos dar uma olhada em alguns dos aplicativos desse gênero e seu número de instalações:

In [51]:
for app in playStory_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

O gênero livro e referência inclui uma variedade de aplicativos: software para processamento e leitura de e-books, várias coleções de bibliotecas, dicionários, tutoriais sobre programação ou linguagens, etc. Parece que ainda há um pequeno número de aplicativos extremamente populares que distorcem a média:

In [52]:
for app in playStory_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


No entanto, parece que existem apenas alguns aplicativos muito populares, então esse mercado ainda mostra potencial. Vamos tentar obter algumas ideias de aplicativos com base nos tipos de aplicativos que estão em algum lugar no meio em termos de popularidade (entre 1.000.000 e 100.000.000 de downloads):

In [53]:
for app in playStory_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Esse nicho parece ser dominado por softwares de processamento e leitura de e-books, além de várias coleções de bibliotecas e dicionários, portanto, provavelmente não é uma boa ideia criar aplicativos semelhantes, pois haverá uma concorrência significativa.

Também notamos que existem alguns aplicativos criados em torno do livro Alcorão, o que sugere que criar um aplicativo em torno de um livro popular pode ser lucrativo. Parece que pegar um livro popular (talvez um livro mais recente) e transformá-lo em um aplicativo pode ser lucrativo para os mercados do Google Play e da App Store.

No entanto, parece que o mercado já está cheio de bibliotecas, então precisamos adicionar alguns recursos especiais além da versão bruta do livro. Isso pode incluir citações diárias do livro, uma versão em áudio do livro, testes sobre o livro, um fórum onde as pessoas podem discutir o livro, etc.


## Conclusões 

Neste projeto, analisamos dados sobre os aplicativos móveis da App Store e Google Play com o objetivo de recomendar um perfil de aplicativo que possa ser lucrativo para ambos os mercados.

Concluímos que pegar um livro popular (talvez um livro mais recente) e transformá-lo em um aplicativo pode ser lucrativo para os mercados do Google Play e da App Store. Os mercados já estão cheios de bibliotecas, então precisamos adicionar alguns recursos especiais além da versão bruta do livro. Isso pode incluir citações diárias do livro, uma versão em áudio do livro, testes sobre o livro, um fórum onde as pessoas podem discutir o livro, etc.